tests/plugins/test_hide_files.py

```diff
@@ -0,0 +1,244 @@
+from atomic_reactor.util import df_parser, ImageName
+
+from tests.constants import SOURCE, MOCK
+from tests.stubs import StubInsideBuilder
+
+
+if MOCK:
+    from tests.docker_mock import mock_docker
+
+
+class TestHideFilesPlugin(object):
+
+    def test_missing_config(self, tmpdir):
+        df_content = dedent("""\
+            FROM fedora
+            RUN yum install -y python-flask
+            CMD /bin/bash
+            """)
+        df = df_parser(str(tmpdir))
+        df.content = df_content
+
+        tasker, workflow = self.prepare(df.dockerfile_path)
+
+        runner = PreBuildPluginsRunner(tasker, workflow, [
+            {'name': HideFilesPlugin.key, 'args': {}},
+        ])
+        runner_results = runner.run()
+
+        assert runner_results[HideFilesPlugin.key] is None
+        # Verify Dockerfile contents have not changed
+        assert df.content == df_content
+
+    @pytest.mark.parametrize('inherited_user', (None, "custom_user"))
+    @pytest.mark.parametrize('user_in_df', (None, "custom_docker_user"))
+    def test_hide_files(self, tmpdir, inherited_user, user_in_df):
+        if user_in_df:
+            df_content = dedent("""\
+                FROM sha256:123456
+                RUN yum install -y python-flask
+                USER {0}
+                CMD /bin/bash
+                """.format(user_in_df))
+        else:
+            df_content = dedent("""\
+                FROM sha256:123456
+                RUN yum install -y python-flask
+                CMD /bin/bash
+                """)
+        df = df_parser(str(tmpdir))
+        df.content = df_content
+        hide_files = {'tmpdir': '/tmp', 'files': ['/etc/yum.repos.d/repo_ignore_1.repo',
+                                                  '/etc/yum.repos.d/repo_ignore_2.repo']}
+        parent_images = {
+            ImageName.parse('fedora'): ImageName.parse("sha256:123456")
+        }
+
+        tasker, workflow = self.prepare(
+            df.dockerfile_path, hide_files=hide_files, parent_images=parent_images,
+            inherited_user=inherited_user)
+
+        runner = PreBuildPluginsRunner(tasker, workflow, [
+            {'name': HideFilesPlugin.key, 'args': {}},
+        ])
+        runner.run()
+
+        if inherited_user and user_in_df:
+            expected_df_content = dedent("""\
+                FROM sha256:123456
+                USER 0
+                RUN mv -f /etc/yum.repos.d/repo_ignore_1.repo /tmp || :
+                RUN mv -f /etc/yum.repos.d/repo_ignore_2.repo /tmp || :
+                USER {0}
+                RUN yum install -y python-flask
+                USER {1}
+                CMD /bin/bash
+                USER 0
+                RUN mv -fZ /tmp/repo_ignore_1.repo /etc/yum.repos.d/repo_ignore_1.repo || :
+                RUN mv -fZ /tmp/repo_ignore_2.repo /etc/yum.repos.d/repo_ignore_2.repo || :
+                USER {1}
+                """.format(inherited_user, user_in_df))
+        elif inherited_user:
```

Comment:
Minor: the different expected strings could be part of the parametrization for a more readable implementation.